### PR TITLE
Workspace

### PR DESCRIPTION
**Pull request by miloszizic**
Custom to official module migration.
Added workspaces.

### CODE REVIEW

File: output.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  value = module.vpc.database_subnets
+}
+
+output "database_subnets_cidr_blocks" {
+  value = module.vpc.database_subnets_cidr_blocks
+}
+output "kubeconfig" {
+  description = "kubectl config as generated by the module."
+  value       = module.eks.kubeconfig
+}
```
Comment (vladimirmukhin) [inline output.tf:39]: It's a good practice not to expose outputs unless it's really required. Consumed by another terraform state, for example

File: workspace/dev/vpc.tfvars [hunk 2]
```diff
@@ -0,0 +1,6 @@
+aws_region         = "us-east-1"
+vpc_cidr    	   = "10.0.0.0/16"
+availability_zones = ["us-east-1a", "us-east-1b"]
```
Comment (vladimirmukhin) [inline workspace/dev/vpc.tfvars:3]: It's better to replace hardcoded availability zones with this:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/availability_zones

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  cluster_name = "kanbak-eks"
+}
+
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+
+  name = "vpc-${terraform.workspace}"
+  cidr = var.vpc_cidr
+
+  azs = var.availability_zones
```
Comment (vladimirmukhin) [inline main.tf:15]: When multiple arguments with single-line values appear on consecutive lines at the same nesting level, align their equals signs.

There is no need to do this manually, just run `terraform fmt -recursive` in the root of the project and commit changes

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	}
+  	]
+  	ingress_with_cidr_blocks = [
+
+	{
+	  from_port   = 22
+	  to_port     = 22
+	  protocol    = "tcp"
+	  description = "ssh"
+	  cidr_blocks = "0.0.0.0/0"
```
Comment (vladimirmukhin) [inline main.tf:53]: It's better not to open SSH to the entire internet. Use your public IP instead

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,97 @@
+
+provider "aws" {
+  region = var.aws_region
+}
+locals {
+  cluster_name = "kanbak-eks"
```
Comment (vladimirmukhin) [inline main.tf:6]: This variable should be workspaced

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	tags = {
+	  Terraform   = "true"
+	  Environment = terraform.workspace
+	}
+
+}
+module "ec2_bastion" {
+  source                 = "terraform-aws-modules/ec2-instance/aws"
+
+  name                   = "Bastion"
```
Comment (vladimirmukhin) [inline main.tf:65]: This variable should be workspaced

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: output.tf (hunk 1 above)
Comment (miloszizic) [reply output.tf]: Fixed

File: workspace/dev/vpc.tfvars (hunk 2 above)
Comment (miloszizic) [reply workspace/dev/vpc.tfvars]: Fixed

File: main.tf (hunk 3 above)
Comment (miloszizic) [reply main.tf]: Fixed

File: main.tf (hunk 4 above)
Comment (miloszizic) [reply main.tf]: Fixed

File: main.tf (hunk 5 above)
Comment (miloszizic) [reply main.tf]: Fixed

File: main.tf (hunk 6 above)
Comment (miloszizic) [reply main.tf]: Fixed

**Pull request merged**